ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,210 @@
+-   click on the JDBC/ODBC tab
+-   in HTTP path, copy everything after the last /. This is your Cluster ID 
 
 ------------------------------------------------------------------------
 
-Once you have set up an ODBC connection as detailed above, you can then use that connection to pull data directly from Databricks into R Studio. Charlotte recorded a video demonstrating two possible methods of how to do this. The recording is embedded below: 
+### Pulling data into RStudio from Databricks
+
+------------------------------------------------------------------------
+
+Now that you have enabled ODBC connections on your laptop, and enabled a connection between Databricks and RStudio, you can add code to your existing scripts to pull data into RStudio for analysis. If you have connected to other SQL databases before, this code will look quite familiar to you. 
+
+To access the data, we will make use of the `ODBC` package. You can find documentation about this package [on the Posit website](https://solutions.posit.co/connections/db/r-packages/odbc/). You will also need to have the `DBI` package installed.  
+
+Include the following code in your R Script:
 
-<div align="center">
-<iframe src="https://educationgovuk.sharepoint.com/sites/lvewp00086/_layouts/15/embed.aspx?UniqueId=5fad039e-a763-40c8-8ea0-403aea712f4c&embed=%7B%22ust%22%3Atrue%2C%22hv%22%3A%22CopyEmbedCode%22%7D&referrer=StreamWebApp&referrerScenario=EmbedDialog.Create" width="640" height="360" frameborder="0" scrolling="no" allowfullscreen title="ADA_Rstudio_2.mp4"></iframe>
-</div>
 
-A template of all of the code used in the above video can be found in the [ADA_RStudio_connect GitHub repo](https://github.com/dfe-analytical-services/ADA_RStudio_connect).
+``` {r databricks_connect_sql, eval=FALSE}
+library(odbc)
+library(DBI)
 
-Key takeaways from the video and example code: 
+con <- DBI::dbConnect(
+  odbc::databricks(),
+  httpPath = Sys.getenv("DATABRICKS_SQL_PATH")
+)
 
-* The main change here compared to connecting to SQL databases is the connection method. The installation and setup of the ODBC driver are all done pre-code, and the only part of the code that will need updating is your connection (usually your con variable). 
-* If your existing code was pulling in tables from SQL via the `RODBC` package or the `dbplyr` package, then this code should in theory run with minimal edits needed. 
-* If you were writing tables back into SQL from R, this is where your code may need the most edits. 
-* If your code is stored in a repo where multiple analysts contribute to and run the code, in order for the code to run for everyone you will all need to individually install the ODBC driver and **give it the same name** so that when the `con` variable is called, the name used in the code matches everyone's individual driver and runs for everyone. If this is the case, please **add a note about this to your repo's readme file** to help your future colleagues.
+odbcListObjects(con)
+```
 
+---
```

Comment:
Do we want to add a section here to show the code for connecting to a personal cluster as well?

Reply:
I see your point, but as things stand we have two separate pages: 
https://dfe-analytical-services.github.io/analysts-guide/ADA/databricks_rstudio_personal_cluster.html
https://dfe-analytical-services.github.io/analysts-guide/ADA/databricks_rstudio_sql_warehouse.html

I think keeping them separate is logical as it avoids confusion, we can give a lot more context info at the start of each page about the purpose of the SQL Warehouse/personal cluster. People will be able to follow the page from start to finish and do their setup, without getting to the end of one section and then realising there's an entire other section below, and not being sure which to make use of. With these pages I'd ideally like to be able to just point people to them, say "follow these instructions" and then they get to the end of the page and they've got what they need.